### PR TITLE
nixos/ksm: rewrite using systemd-tmpfiles

### PR DESCRIPTION
**Pull request by rnhmjoj**
###### Motivation for this change

This is to get rid of the ugly systemd-udev-settle dependency.
See #73095 for more information.

###### Things done

- [x] Tested via one or more NixOS test (nixosTests.ksm)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by flokli** [comment]
How would moving this to `tmpfiles` fix the problem? This seems to be a script writing to `/sys`, possibly requiring these files to be present (and underlying hardware to be appropriately initialized), hence the dependency on udevadm-settle.

Nothing guarantees systemd-tmpfiles runs "late enough" for this to has happened.

Can this be some udev rule instead?

**Comment by flokli** [comment]
RedHat seems to have some daemon here, provided by qemu: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/virtualization_tuning_and_optimization_guide/chap-ksm

**Comment by rnhmjoj** [comment]
> How would moving this to tmpfiles fix the problem?

It changes nothing in this regard, but I don't think there is a problem to begin with, at least not anymore. I don't see why `/sys/kernel/mm/ksm/` shouldn't be available as soon as the kernel is executed.

Try building a system with:
```
boot.initrd.preDeviceCommands = ''
 echo 1 > /sys/kernel/mm/ksm/run; cat /sys/kernel/mm/ksm/run
'';
```
this runs in stage 1, before udev is even started and succeeds.

Btw, I stole the tmpfiles rule from https://wiki.archlinux.org/index.php/QEMU#Enabling_KSM.

**Comment by flokli** [comment]
In that case, I'd prefer if this could stay a shell script, and we just drop the dependency. If we do that, worst case the script will fail (and people will realize). If this is switched to a tmpfiles rule, it might do the right thing every second time, and every other time it'll try to create the underlying file structure by itself ;-)

**Comment by rnhmjoj** [comment]
> and every other time it'll try to create the underlying file structure by itself ;-)

According to `tmpfiles.d(5)` this should never happen: if the file does not exist it should simply fail, but ok I'll keep the script.

**Review by flokli — APPROVED** [review]

File: nixos/tests/ksm.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import ./make-test-python.nix ({ pkgs, ...} :
+
+{
+  name = "ksm";
+  meta = with pkgs.lib.maintainers; {
```
Comment (flokli) [inline nixos/tests/ksm.nix:5]:
```suggestion
import ./make-test-python.nix ({ lib, ...} :

{
  name = "ksm";
  meta = with lib.maintainers; {
```

**Review by flokli — CHANGES_REQUESTED** [review]
don't use`pkgs.lib`, but `lib` directly

File: nixos/tests/ksm.nix (hunk 1 above)
Comment (rnhmjoj) [reply nixos/tests/ksm.nix]: Thanks, I never noticed it (pasted this from an old template).

**Pull request merged**